jdaviz/configs/specviz/plugins/line_analysis/line_analysis.py

```diff
@@ -436,6 +436,11 @@ def _uncertainty(result):
 
         temp_results = []
         spec_subtracted = spectrum - continuum
+
+        if spec_subtracted.mask is not None:
+            # temporary fix while mask may contain None:
+            spec_subtracted.mask = spec_subtracted.mask.astype(bool)
```

Comment:
what does this do to the bools and is that consistent with what they're _supposed_ to mean?

Reply:
It leaves the bools as bools, as far as I can tell.

Reply:
Interestingly, doing `arr.dtype` on a mix of None, True and False gives `dtype('O')`, as opposed to `dtype('bool')` for just True and False.

Reply:
ok, I meant what does this do to `Nones`

Reply:
In `np.ma.array`, `None` is silently converted to `False`. In that context, `None` means *not masked*.

Reply:
Summarized above in the description: 
> Casting to bool dodges errors by interpreting None as False.

Reply:
So this is what happens when you treat it as plain numpy array, which I think is correct?

```python
>>> np.array([True, False, None]).astype(bool)
array([ True, False, False])
```

Reply:
Yep.